### PR TITLE
AsmHentai - new source

### LINKED ISSUE #1016: Asm hentai source request!
### Source name

Asm Hentai

### Source link

https://asmhentai.com

### Source language

English, Multi

### Other details

18+ - NSFW = Yes

Its so good and have lots of hen!!

### Acknowledgements

- [X] I have checked that the extension does not already exist by searching the [extension listing](https://keiyoushi.github.io/extensions/) and verified it does not appear there.
- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a meaningful title with the source name.
- [ ] I have admitted that I am a clown by having checked this box, as I have not read these acknowledgements.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by SnakeDoc83**
CLoses #1016 

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [x] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [x] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

File: src/all/asmhentai/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+                <category android:name="android.intent.category.DEFAULT" />
+                <category android:name="android.intent.category.BROWSABLE" />
+                <data
+                    android:host="asmhentai.com"
+                    android:pathPattern="/g/..*"
+                    android:scheme="https" />
+            </intent-filter>
+        </activity>
+    </application>
+</manifest>
```
Comment (AwkwardPeak7) [inline src/all/asmhentai/AndroidManifest.xml]: newline
```suggestion
</manifest>

```

File: src/all/asmhentai/build.gradle [hunk 2]
```diff
@@ -0,0 +1,8 @@
+ext {
+    extName = 'AsmHentai'
+    extClass = '.ASMHFactory'
+    extVersionCode = 1
+    isNsfw = true
+}
+
+apply from: "$rootDir/common.gradle"
```
Comment (AwkwardPeak7) [inline src/all/asmhentai/build.gradle]:
```suggestion
apply from: "$rootDir/common.gradle"

```

File: src/all/asmhentai/src/eu/kanade/tachiyomi/extension/all/asmhentai/AsmHentai.kt [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+                .add("_token", token)
+                .add("id", document.inputIdValueOf("load_id"))
+                .add("dir", document.inputIdValueOf("load_dir"))
+                .add("visible_pages", "10")
+                .add("t_pages", totalPages)
+                .add("type", "2") // 1 would be "more", 2 is "all remaining"
+                .build()
+
+            val xhrHeaders = headers.newBuilder()
+                .add("Content-Type", "application/x-www-form-urlencoded; charset=UTF-8")
```
Comment (AwkwardPeak7) [inline src/all/asmhentai/src/eu/kanade/tachiyomi/extension/all/asmhentai/AsmHentai.kt]: okhttp will automatically set Content-Type header based on request body
```suggestion
```

Comment (SnakeDoc83) [reply src/all/asmhentai/src/eu/kanade/tachiyomi/extension/all/asmhentai/AsmHentai.kt]: Sure

**Review by beer-psi — APPROVED** [review]

**Pull request merged**